hal/src/main/native/sim/MockData/SPIData.cpp

```diff
@@ -13,6 +13,21 @@
 
 using namespace hal;
 
+void InvokeCallback(
+    std::shared_ptr<SpiAutoReceiveDataListenerVector> currentVector,
+    const char* name, uint8_t* buffer, int32_t numToRead,
+    int32_t* outputCount) {
+  // Return if no callbacks are assigned
+  if (currentVector == nullptr) return;
```

Comment:
Could you add a newline after this? Semantic whitespace makes it easier to visually parse things and provides hints on what lines a comment applies to (the latter isn't an issue for this).

Reply:
Formatter pops it back

Reply:
Ok. That's annoying.